lib/dialects/snowflake/query-generator.js

```diff
@@ -0,0 +1,633 @@
+        // true or false cannot be casted as booleans within a JSON structure
+        smth.type = 'char';
+      } else if (/double precision/i.test(smth.type) || /boolean/i.test(smth.type) || /integer/i.test(smth.type)) {
+        smth.type = 'decimal';
+      } else if (/text/i.test(smth.type)) {
+        smth.type = 'char';
+      }
+    }
+
+    return super.handleSequelizeMethod(smth, tableName, factory, options, prepend);
+  }
+
+  /*
+  _toJSONValue(value) {
+    // true/false are stored as strings in mysql
+    if (typeof value === 'boolean') {
+      return value.toString();
+    }
+    // null is stored as a string in mysql
+    if (value === null) {
+      return 'null';
+    }
+    return value;
+  }
+  */
+
+  truncateTableQuery(tableName) {
+    return `TRUNCATE ${this.quoteTable(tableName)}`;
+  }
+
+  deleteQuery(tableName, where, options = {}, model) {
+    const table = this.quoteTable(tableName);
+    let whereClause = this.getWhereConditions(where, null, model, options);
+    const limit = options.limit ? ` LIMIT ${this.escape(options.limit)}` : '';
+    let primaryKeys = '';
+    let primaryKeysSelection = '';
+
+    if (whereClause) {
+      whereClause = ` WHERE ${whereClause}`;
+    }
+
+    if (options.limit) {
+      if (!model) {
+        throw new Error('Cannot LIMIT delete without a model.');
+      }
+
+      const pks = Object.values(model.primaryKeys).map(pk => this.quoteIdentifier(pk.field)).join(',');
+
+      primaryKeys = model.primaryKeyAttributes.length > 1 ? `(${pks})` : pks;
+      primaryKeysSelection = pks;
+
+      return `DELETE FROM ${table} WHERE ${primaryKeys} IN (SELECT ${primaryKeysSelection} FROM ${table}${whereClause}${limit})`;
+    }
+    return `DELETE FROM ${table}${whereClause}`;
+  }
+
+  showIndexesQuery() {
+    return 'SELECT \'\' FROM DUAL';
+  }
+
+  showConstraintsQuery(table, constraintName) {
+    const tableName = table.tableName || table;
+    const schemaName = table.schema;
+
+    return Utils.joinSQLFragments([
+      'SELECT CONSTRAINT_CATALOG AS constraintCatalog,',
+      'CONSTRAINT_NAME AS constraintName,',
+      'CONSTRAINT_SCHEMA AS constraintSchema,',
+      'CONSTRAINT_TYPE AS constraintType,',
+      'TABLE_NAME AS tableName,',
+      'TABLE_SCHEMA AS tableSchema',
+      'from INFORMATION_SCHEMA.TABLE_CONSTRAINTS',
+      `WHERE table_name='${tableName}'`,
+      constraintName && `AND constraint_name = '${constraintName}'`,
+      schemaName && `AND TABLE_SCHEMA = '${schemaName}'`,
+      ';'
+    ]);
+  }
+
+  removeIndexQuery(tableName, indexNameOrAttributes) {
+    let indexName = indexNameOrAttributes;
+
+    if (typeof indexName !== 'string') {
+      indexName = Utils.underscore(`${tableName}_${indexNameOrAttributes.join('_')}`);
+    }
+
+    return Utils.joinSQLFragments([
+      'DROP INDEX',
+      this.quoteIdentifier(indexName),
+      'ON',
+      this.quoteTable(tableName)
+    ]);
+  }
+
+  attributeToSQL(attribute, options) {
```

Comment:
`Utils.joinSQLFragments` could also be used here

Reply:
`attributeToSQL` better to stay in existing...it is a bit complex